src/render/draw_hillshade.js

```diff
@@ -42,7 +42,7 @@ function drawHillshade(painter: Painter, sourceCache: SourceCache, layer: Hillsh
 function renderHillshade(painter, tile, layer, depthMode, stencilMode, colorMode) {
     const context = painter.context;
     const gl = context.gl;
-    const fbo = tile.fbo;
```

Comment:
@arindam1993 
I might have misunderstood the purpose of cache here.
The patch is introducing the cache in order to simplify evicting (when max size limit change) procedure - when there is need to remove content from cache, it is easier to go through cache.
This comes for a price we have to pay in every render call - instead of `tile.fbo`, there is `painter.getFboForTile('dem', tile.tileID);`. Note that similar would need to be also port to native.

Could we keep the previous and for evicting enumerate tiles in DEM sourceCache and remove tiles?

Reply:
The purpose of the cache was to manage allocations and deallocations of per-tile GPU textures and FBO's.
Currently it's handled by the source cache, which can have an unnecessarily large number of cached tiles, some of which have been offscreen for a long time, so the idea was to have something in the middle. I've changed the design a bit now, so its a bit more temporal as well.